src/main/groovy/io/openliberty/tools/gradle/utils/LooseWarApplication.groovy

```diff
@@ -1,27 +1,34 @@
 package io.openliberty.tools.gradle.utils;
 
-import java.io.File;
-import org.gradle.api.Project;
-import org.gradle.api.plugins.WarPluginConvention
+import io.openliberty.tools.common.plugins.config.LooseApplication
+import io.openliberty.tools.common.plugins.config.LooseConfigData
 import org.gradle.api.Task
+import org.gradle.api.logging.Logger
 import org.gradle.api.tasks.bundling.War
 
-import io.openliberty.tools.common.plugins.config.LooseConfigData
-import io.openliberty.tools.common.plugins.util.PluginExecutionException
-import io.openliberty.tools.common.plugins.config.LooseApplication
-
 public class LooseWarApplication extends LooseApplication {
     
     protected Task task;
+    protected Logger logger;
 
-    public LooseWarApplication(Task task, LooseConfigData config) {
-        super(task.getProject().getBuildDir().getAbsolutePath(), config)
+    public LooseWarApplication(Task task, LooseConfigData config, Logger logger) {
+        super(task.getProject().getLayout().getBuildDirectory().getAsFile().get().getAbsolutePath(), config)
         this.task = task
+        this.logger = logger
     }
 
     public void addSourceDir() throws Exception {
-        WarPluginConvention wpc = task.getProject().getConvention().findPlugin(WarPluginConvention)
-        File sourceDir = new File(wpc.getWebAppDir().getAbsolutePath())
+
+        War war;
+        File sourceDir = new File("src/main/webapp")
+        if (task.getProject().getPlugins().hasPlugin("war")) {
+            war = (War) task.getProject().war
+            if (war.getWebAppDirectory().getAsFile().get() != null) {
+                sourceDir = new File(war.getWebAppDirectory().getAsFile().get().getAbsolutePath())
+            }
+        }else {
+            logger.warn("Default sourcedir path to src/main/webapp as WAR plugin does not exist. Application may not work expectedly")
```

Comment:
"Could not get the webAppDirectory location from the WAR plugin because it is not configured. Using the default location src/main/webapp instead. Application may not work as expected."

Reply:
Done